lib/onelogin/ruby-saml/logoutresponse.rb

```diff
@@ -69,46 +104,211 @@ def status_code
-          return soft ? false : validation_error("Blank response")
+        if response.nil? or response.empty?
+          error_msg = "Blank Logout Response"
+          @errors << error_msg
+          return soft ? false : validation_error(error_msg)
         end
 
         if settings.nil?
-          return soft ? false : validation_error("No settings on response")
+          error_msg = "No settings on Logout Response"
+          @errors << error_msg
+          return soft ? false : validation_error(error_msg)
         end
 
         if settings.issuer.nil?
-          return soft ? false : validation_error("No issuer in settings")
+          error_msg = "No issuer in settings"
+          @errors << error_msg
+          return soft ? false : validation_error(error_msg)
         end
 
         if settings.idp_cert_fingerprint.nil? && settings.idp_cert.nil?
-          return soft ? false : validation_error("No fingerprint or certificate on settings")
+          error_msg = "No fingerprint or certificate on settings"
+          @errors << error_msg
+          return soft ? false : validation_error(error_msg)
+        end
+
+        true
+      end
+
+      # Validates the Logout Response against the specified schema.
+      # If fails, the error is added to the errors array
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @return [Boolean] True if the XML is valid, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #
+      def validate_structure(soft = true)
+        begin 
+          valid = valid_saml?(document, soft)
+          unless valid
+            @errors << "Invalid Logout Response. Not match the saml-schema-protocol-2.0.xsd"
+          end
+          valid
+        rescue OneLogin::RubySaml::ValidationError => e
+          @errors << e.message
+          raise e
+        end
+      end
+
+      # Validates the Status of the Logout Response
+      # If fails, the error is added to the errors array, including the StatusCode returned and the Status Message.
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the response is invalid or not)      
+      # @return [Boolean] True if the Logout Response contains a Success code, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #
+      def validate_success_status(soft = true)
+        return true if success?
+
+        error_msg = 'The status code of the Logout Response was not Success'
+        unless status_code.nil?
+          printable_code = status_code.split(':').pop
+          error_msg +=  ', was ' + printable_code
+        end
+
+        unless status_message.nil?
+          error_msg +=  ' -> ' + status_message
+        end
+
+        @errors << error_msg
+        soft ? false : validation_error(error_msg)
```

Comment:
Most places in the changeset use an explicit return for lines like these. While I prefer the implicit return style, I'd rather see consistent usage across this library.